services/api/tests/test_publish_persona_check.py

```
@@ -0,0 +1,449 @@
+def test_non_httpx_exception_from_auth_chain_still_records_honest_verdict(monkeypatch):
+    """Runtime-trace finding on PR #562: a token-exchange RuntimeError, an
+    SSM botocore error, or a malformed-JSON response are all real, reachable
+    failures from the publish boundary that are NOT httpx.HTTPStatusError/
+    RequestError - the except must be total, not HTTP-shaped, or these
+    silently skip record_check_verdict entirely."""
+    from personas import publish_check
+
+    recorded = {}
+    monkeypatch.setattr(
+        publish_check, "with_install_token_retry",
+        lambda *a, **kw: (_ for _ in ()).throw(RuntimeError("malformed token response")),
```

Ruff (0.15.20)

[warning] Unused lambda argument: `a`

(ARG005)